crates/hir_ty/src/diagnostics/expr.rs

```diff
@@ -206,6 +244,26 @@ impl<'a, 'b> ExprValidator<'a, 'b> {
             }
         }
 
+        let mut args_with_params = args.iter().zip(params.iter());
+
+        if is_method_call {
+            args_with_params.next();
+        }
+
+        args_with_params
+            .filter_map(|(arg, param)| {
+                let (arg, mutability) = check_missing_refs(infer, *arg, param)?;
```

Comment:
Hm, I still don't understand why we need to single out call expresisons here? It seems better to just put the logic here to `validate_body`?

This diagnostic can fire for *any* expressions, as long as `TypeMismatch` records types differing only in referenceness.

Reply:
I probably got it, tests are still passing